cmd/scan/framework.go

```diff
@@ -121,39 +125,73 @@ func getFrameworkCmd(ks meta.IKubescape, scanInfo *cautils.ScanInfo) *cobra.Comm
+		{reporthandlingapis.SeverityHighString, severityCounters.NumberOfResourcesWithHighSeverity},
+		{reporthandlingapis.SeverityCriticalString, severityCounters.NumberOfResourcesWithCriticalSeverity},
+	}
+
+	targetSeverityIdx := 0
+	for idx, description := range getFailedResourcesFuncsBySeverity {
+		if strings.ToLower(description.SeverityName) == strings.ToLower(targetSeverity) {
+			targetSeverityIdx = idx
+			break
+		}
+	}
+
+	for _, description := range getFailedResourcesFuncsBySeverity[targetSeverityIdx:] {
+		failedResourcesCount := description.GetFailedResources()
+		if failedResourcesCount > 0 {
+			return true, nil
+		}
+	}
+
+	return false, nil
+
+}
+
+// enforceSeverityThresholds ensures that the scan results are below the defined severity threshold
 //
-// The function forces the application to terminate with an exit code 1 if there are more resources with failed controls of a given severity than permitted
+// The function forces the application to terminate with an exit code 1 if at least one control failed control that exceeds the set severity threshold
 func enforceSeverityThresholds(severityCounters reportsummary.ISeverityCounters, scanInfo *cautils.ScanInfo) {
-	failedCritical := severityCounters.NumberOfResourcesWithCriticalSeverity()
-	failedHigh := severityCounters.NumberOfResourcesWithHighSeverity()
-	failedMedium := severityCounters.NumberOfResourcesWithMediumSeverity()
-	failedLow := severityCounters.NumberOfResourcesWithLowSeverity()
-
-	criticalExceeded := failedCritical > scanInfo.FailThresholdCritical
-	highExceeded := failedHigh > scanInfo.FailThresholdHigh
-	mediumExceeded := failedMedium > scanInfo.FailThresholdMedium
-	lowExceeded := failedLow > scanInfo.FailThresholdLow
-
-	resourceThresholdsExceeded := criticalExceeded || highExceeded || mediumExceeded || lowExceeded
-
-	if resourceThresholdsExceeded {
-		logger.L().Fatal(
-			"There were failed controls that exceed permitted severity thresholds",
-			helpers.String("critical", fmt.Sprintf("got: %d, permitted: %d", failedCritical, scanInfo.FailThresholdCritical)),
-			helpers.String("high", fmt.Sprintf("got: %d, permitted: %d", failedHigh, scanInfo.FailThresholdHigh)),
-			helpers.String("medium", fmt.Sprintf("got: %d, permitted: %d", failedMedium, scanInfo.FailThresholdMedium)),
-			helpers.String("low", fmt.Sprintf("got: %d, permitted: %d", failedLow, scanInfo.FailThresholdLow)),
-		)
+	if val, err := countersExceedSeverityThreshold(severityCounters, scanInfo); val && err == nil {
+		logger.L().Fatal("one of the failed controls exceeds the severity threshold", helpers.String("set severity threshold", scanInfo.FailThresholdSeverity))
+	} else if err != nil {
+		logger.L().Fatal(err.Error())
 	}
 }
 
+// validateSeverity returns an error if a given severity is not known, nil otherwise
+func validateSeverity(severity string) error {
+	for _, val := range reporthandlingapis.GetSupportedSeverities() {
```

Comment:
This part requires a new version of OPA utils: kubescape/opa-utils#35

Reply:
Fixed by [16944ba](https://github.com/kubescape/kubescape/pull/838/commits/16944ba8c99e830a36b3cd73723419e60bea8d0d).